bcipy/acquisition/protocols/lsl/lsl_client.py

```diff
@@ -126,12 +139,34 @@ def __exit__(self, _exc_type, _exc_value, _traceback):
         """Context manager exit method to clean up resources."""
         self.stop_acquisition()
 
+    def _data_stats(self, data: List[Record]) -> Dict[str, float]:
+        """Summarize a list of records for logging and inspection."""
+        if data:
+            diffs = pd.DataFrame(data)['timestamp'].diff()
+            data_start = data[0].timestamp
+            data_end = data[-1].timestamp
+            precision = 3
+            return {
+                'count': len(data),
+                'seconds': round(data_end - data_start, precision),
+                'from': round(data_start, precision),
+                'to': round(data_end, precision),
+                'expected_diff': round(1 / self.device_spec.sample_rate,
+                                       precision),
+                'mean_diff': round(diffs.mean(), precision),
+                'max_diff': round(diffs.max(), precision)
+            }
+        return {}
```

Comment:
Would it make sense to return none? Then a caller could check for that condition instead of failing on a key error.

Reply:
My primary use case for the dict is to print out the whole thing. If it might be a `None` value I would need some additional checks. I think in this case if we're not sure that a key would be in the map it would be cleaner to use a `map.get(key, default_value)`.